internal/prompter/test.go

```diff
@@ -29,3 +30,258 @@ func NoSuchAnswerErr(answer string) error {
 func NoSuchPromptErr(prompt string) error {
 	return fmt.Errorf("no such prompt '%s'", prompt)
 }
+
+type SelectStub struct {
+	Prompt       string
+	ExpectedOpts []string
+	Fn           func(string, string, []string) (int, error)
+}
+
+type InputStub struct {
+	Prompt string
+	Fn     func(string, string) (string, error)
+}
+
+type ConfirmStub struct {
+	Prompt string
+	Fn     func(string, bool) (bool, error)
+}
+
+type MultiSelectStub struct {
+	Prompt       string
+	ExpectedOpts []string
+	Fn           func(string, string, []string) ([]string, error)
+}
+
+type InputHostnameStub struct {
+	Fn func() (string, error)
+}
+
+type PasswordStub struct {
+	Prompt string
+	Fn     func(string) (string, error)
+}
+
+type AuthTokenStub struct {
+	Fn func() (string, error)
+}
+
+type ConfirmDeletionStub struct {
+	Prompt string
+	Fn     func(string) error
+}
+
+type MockPrompter struct {
+	PrompterMock
+	t                    *testing.T
+	SelectStubs          []SelectStub
+	InputStubs           []InputStub
+	ConfirmStubs         []ConfirmStub
+	MultiSelectStubs     []MultiSelectStub
+	InputHostnameStubs   []InputHostnameStub
+	PasswordStubs        []PasswordStub
+	AuthTokenStubs       []AuthTokenStub
+	ConfirmDeletionStubs []ConfirmDeletionStub
+}
+
+// TODO thread safety
```

Comment:
maybe it's okay if these methods aren't threadsafe, prompts in the UI cannot happen in parallel anyway?

Reply:
ah good point